privacyidea/api/auth.py

```diff
@@ -226,23 +226,24 @@ def get_auth_token():
         raise AuthError(_("Authentication failure. Missing Username"),
                         id=ERROR.AUTHENTICATE_MISSING_USERNAME)
 
-    loginname, realm = split_user(username)
```

Comment:
Why overcomplicating things and not simply to a
~~~~Python
    loginname, realm = split_user(username)
    realm = realm.lower()
~~~~

or adding a `lower()` to split_username.
Verifying all the changes and ensuring no side effect, is near to impossible.

What is the reason for moving the creation of a user object around further below in the code?

Reply:
Because the user object is initially created in the `before_request()` and added to the request object here:
https://github.com/privacyidea/privacyidea/blob/fa643349df8fabdf329caf4091b70eaafaade74a/privacyidea/api/auth.py#L120
This way we can remove a second ldap request when the user object is created again later.
But we can move this change to a separate issue and just tackle the case-sensitivity here.

Reply:
> This way we can remove a second ldap request when the user object is created again later.

Ok, I understand. Will look into it again.